corehq/apps/auditcare/models.py

```diff
@@ -125,6 +126,11 @@ class NavigationEventAudit(AuditEvent):
     headers = NullJsonField(default=dict)
     status_code = models.SmallIntegerField(default=0)
 
+    class Meta:
+        indexes = AuditEvent._meta.indexes + [
+            models.Index(fields=['couch_id'], name="audit_nav_couch_875bc_idx"),
```

Comment:
I think it would make sense to use a [partial index](https://www.postgresql.org/docs/9.6/indexes-partial.html) here, where rows with `couch_id IS NULL` are not indexed.

> This reduces the size of the index, which will speed up those queries that do use the index. It will also speed up many table update operations because the index does not need to be updated in all cases.

Over time as we archive or drop data from SQL that index will get smaller and smaller. Here's [an example](https://github.com/dimagi/commcare-hq/blob/fb339b841c8a2ed2c9ddcb9e754de02f42cef755/corehq/blobs/models.py#L76-L80) of how to do that in Django as well as an [example of SQL syntax](https://github.com/dimagi/commcare-hq/blob/fb339b841c8a2ed2c9ddcb9e754de02f42cef755/corehq/blobs/migrations/0010_auto_20191023_0938.py#L5-L10).

The same applies to the `AccessAudit` index.

Reply:
Would that also decrease the amount of time to create the index during the migration? I imagine it still has to scan all rows, but they're all `null` to start with, so it doesn't actually have to write anything?

Reply:
Yes, but I'm not sure by how much.

Reply:
Would we not just drop the index once the migration is complete?

Reply:
We could do that. I still think it makes sense to use a partial index to minimize performance impact on new auditcare writes.

Just thought of another thing: the index should be a `UNIQUE` index. This will make it much easier to insert rows while ignoring duplicates.

Reply:
Added 105f9cc71cc but I'm not quite sure if it's correct - not sure if I need to update the models or if that doesn't matter because I'm providing the sql for index creation.